Add luaj and parse lua files & bind lua funtions in java & fix errors in build.gradle & make proto as a subproject

RimuruChan opened the pull request:
Add luaj and parse lua files
Bind lua funtions in java
Fix task jar pack proto files into jar
Make proto as a subproject

KawaiiNahida committed:
[Script] SceneBase

Commit:
rename & format & optimize

Commit:
fix errors

KawaiiNahida committed:
parse SceneGroups

KawaiiNahida committed:
Rename SceneGroup -> SceneBlock

Commit:
recode

Commit:
Merge remote-tracking branch 'origin/development' into development

KawaiiNahida committed:
Create SceneGroup.java

Commit:
recode every shit code

KawaiiNahida committed:
constants that is used in script

Melledy commented:
@RimuruChan I'm working on my implementation already, will be pushing mine in a bit.

Commit:
make proto as a subproject

Commit:
Merge remote-tracking branch 'origin/development' into development

RimuruChan committed:
Merge branch 'Grasscutters:development' into development

Commit:
fix missing proto

Commit:
suppress javadoc warnings
fix gradle warnings

Commit:
suppress javadoc warnings
fix gradle warnings

Commit:
format type files

KawaiiNahida committed:
Sort Const Values & ConstGlobals LuaLib

KawaiiNahida committed:
Create CommonScript.java

KawaiiNahida committed:
[PlaceHolder] ScriptLib apis

GetRegionEntityCountImpl AddQuestProgressImpl KillGroupEntityImpl GetServerTimeImpl GetGroupVariableValueImpl

KawaiiNahida committed:
Add CommonScript to Globals

KawaiiNahida committed:
[Refactor] Move ScriptLib function implementation to scriptlib package

KawaiiNahida committed:
RefreshGroupImpl

KawaiiNahida committed:
PrintLogImpl placeholder
